stress_test/controller.py

```diff
@@ -0,0 +1,378 @@
+        :type test_config: JSON configuration dictionary
+        :type ctrl_base_dir: str
+        """
+        self.name = test_config['controller_name']
+        self.base_dir = ctrl_base_dir
+
+        self.ip = test_config['controller_node_ip']
+        self.ssh_port = test_config['controller_node_ssh_port']
+        self.ssh_user = test_config['controller_node_username']
+        self.ssh_pass = test_config['controller_node_password']
+
+        self.need_rebuild = test_config['controller_rebuild']
+        self.need_cleanup = test_config['controller_cleanup']
+        self.of_port = test_config['controller_port']
+        self.logs_dir = self.base_dir + test_config['controller_logs_dir']
+
+        self.build_hnd = self.base_dir + test_config['controller_build_handler']
+        self.start_hnd = self.base_dir + test_config['controller_start_handler']
+        self.stop_hnd = self.base_dir + test_config['controller_stop_handler']
+        self.status_hnd = self.base_dir + test_config['controller_status_handler']
+        self.clean_hnd = self.base_dir + test_config['controller_clean_handler']
+
+        self.status = 'UNKNOWN'
+
+        self.java_opts = ' '.join(test_config['java_opts'])
+        self.pid = -1
+
+        self._ssh_conn = None
+
+        #check handlers' validity
+        util.file_ops.check_filelist([self.build_hnd,
+                                      self.start_hnd,
+                                      self.stop_hnd,
+                                      self.status_hnd,
+                                      self.clean_hnd])
+
+
+
+    @staticmethod
+    def new (ctrl_base_dir, test_config):
+        """ Factory method. Creates a subclass class depending on the controller name
+        :returns: a subclass or None
+        :rtype: failed_flow_ops int
+        """
+        name = test_config['controller_name']
+        if (name == 'ODL'):
+           return ODL(ctrl_base_dir, test_config)
+
+        elif name == 'ONOS':
+
+           raise NotImplementedError('ONOS is not supported yet')
+        #  return ONOS(ctrl_base_dir, test_config)
+
+        else:
+            raise NotImplementedError('Not supported yet')
+        #   return None
+
+
+    def init_ssh(self):
+        logging.info(
+            '[open_ssh_connection] Initiating SSH session with {0} node.'.
+            format(self.name, self.ip))
+
+        self._ssh_conn = util.netutil.ssh_connect_or_return2(self.ip,
+                                      int(self.ssh_port),
+                                      self.ssh_user,
+                                      self.ssh_pass,
+                                      10)
+
+    def cleanup(self):
+        """Wrapper to the controller cleanup handler
+        """
+        logging.info('[Controller] Cleaning up')
+
+        self.status = 'CLEANING'
+        util.netutil.ssh_run_command(self._ssh_conn,
+                                    ' '.join([self.clean_hnd]),
```

Comment:
(delayed comment)

wouldn't it be much simpler if we give the 2nd argument as just `self.clean_hnd`? 

using `' '.join(...)` is meaningful only if we have multiple strings to join, but here we have just one

check this in other call sites of `ssh_run_command`, too